src/main/java/org/generationcp/middleware/api/germplasm/search/GermplasmSearchServiceImpl.java

```diff
@@ -28,4 +33,54 @@ public GermplasmSearchServiceImpl(final HibernateSessionProvider sessionProvider
 	public long countSearchGermplasm(final GermplasmSearchRequest germplasmSearchRequest, final String programUUID) {
 		return this.daoFactory.getGermplasmSearchDAO().countSearchGermplasm(germplasmSearchRequest, programUUID);
 	}
+
+	@Override
+	public List<UserDefinedField> getGermplasmAttributeTypes(final GermplasmSearchRequest germplasmSearchRequest) {
+		return this.daoFactory.getGermplasmSearchDAO().getGermplasmAttributeTypes(germplasmSearchRequest);
+	}
+
+	@Override
+	public List<UserDefinedField> getGermplasmNameTypes(final GermplasmSearchRequest germplasmSearchRequest) {
+		return this.daoFactory.getGermplasmSearchDAO().getGermplasmNameTypes(germplasmSearchRequest);
+	}
+
+	@Override
+	public Map<Integer, Map<Integer, String>> getGermplasmAttributeValues(final GermplasmSearchRequest germplasmSearchRequest) {
+		final List<Attribute> attributes = this.daoFactory.getGermplasmSearchDAO().getGermplasmAttributeValues(germplasmSearchRequest);
+
+		if (attributes.isEmpty()) {
+			return null;
```

Comment:
It is not a good practice to return null, it will force to validate NPE whenever this function is used. 
Remove the if block, so a new HashMap will be returned